02-peer-protocol.md

```diff
@@ -995,7 +995,10 @@ is destined, is described in [BOLT #4](04-onion-routing.md).
     1. type: 0 (`blinding_point`)
     2. data:
         * [`point`:`blinding`]
-
+    1. type: 1 (`endorsed`)
```

Comment:
I still think this proposal also works without the `endorsed` flag, similarly how current circuitbreaker users are restricting peers based on observed behavior only.

Reply:
Endorsement is important to stop malcious nodes from leveraging honest nodes reputation to spam downstream peers. Take the example of: 
* `A -- B -- C -- D` 
* `A` is a malicious node that has recently opened a channel to `B`, and has neutral reputation with `B`
* `B` is an honest node that has built up good reputation with `C` over time

Without endorsement, `A` can immediately spam payments through `A -- B -- C -- D`, and `C` will start to penalize `B` for the spammy payment flow. 

With endorsement, `B` will set `endorsed=0` for `A`'s payments, so `A` cannot do any damage to `B`'s reputation with `C` without building up good reputation with `B` first.

Without endorsed signalling nodes may be generally incentivised to drop HTLCs (which are actually honest, they just don’t know it) to protect their reputation, which is bad for payment reliability.

Reply:
In this example, can't B just monitor what they receive from A and if it doesn't look right, make sure A is appropriately restricted early on? I think you can argue that it is B's responsibility to ensure that they don't forward bad traffic, and B can make this happen however they think is best.

Reply:
With locally tracked reputation, you'll always be able to "spread out" bad behavior across multiple nodes (below whatever threshold they set) to target a single node - difficult to visualize in text (see 38-57 [here](https://docs.google.com/presentation/d/1feMupT9_k5DeFvwJqe0pcrPisCBgveowQYpSKOWk7hM/edit?usp=sharing)).

>  B's responsibility to ensure that they don't forward bad traffic, and B can make this happen however they think is best

I think that it's ineviatble that we can't _always_ identify "bad" traffic - we'll be uncertain about payments from brand new peers, for example. I think `endorsed` is important for that gray area where we still want LN to be able to service as many payments as possible, but nodes may not be incentivised to risk it.

Reply:
> - `A -- B -- C -- D`
> - `A` is a malicious node that has recently opened a channel to B, and has neutral reputation with B
> - `B` is an honest node that has built up good reputation with C over time

I think the only decent attack option for A is to build up good reputation with B and then decide when to abuse it in order to jam the `B -- C` channel. Worth noting that this involves `B` collecting enough revenue from `A` to allow them into the `endorsed` slots.

> In this example, can't B just monitor what they receive from A and if it doesn't look right, make sure A is appropriately restricted early on?

I think the issue here is that we can't define what "looks right" means.